bazel/example/WORKSPACE.bazel

```diff
@@ -3,6 +3,12 @@ workspace(name = "example_p4_project")
 local_repository(
   name = "com_github_p4lang_p4c",
   path = "../..",
+  # This part is optional and only needed for custom backends with IR
+  # extensions.
+  repo_mapping = {
+    # Tells p4c where to look for `:ir_extensions` target: in this project.
+    "@com_github_p4lang_p4c_extension" : "@example_p4_project",
```

Comment:
pretty neat way to resolve that circular dependency :)

Reply:
@smolkaj BTW is `repo_mapping` supported for `http_archive` and `git_repository`?
It does not say so in the documentation, but this seems to imply it is supported: https://github.com/bazelbuild/bazel/issues/6845. I believe the documentation is just inaccurate?

Reply:
I'll try this tomorrow.

Probs for finding this! I'm not sure yet if this is a really great idea or horribly undermining Bazel concepts :D

Reply:
It's worth noting that `repo_mapping` is not needed to do this. Something like `new_local_repository(name = "com_github_p4lang_p4c_extension", ...)` should also do the trick, though a bit less elegantly. I'll document this tomorrow since it looks like `repo_mapping` might indeed not be available for `http_archive` and `git_repository`.

Reply:
The mapping scheme seems to work!
Currently solving linker errors related to the move from gmp to boost::multiprecision

Reply:
Awesome! :)

Reply:
I also just confirmed that `repo_mapping` works for `http_archive` and `git_repository` as well, so we can leave it as is.